app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -448,6 +460,18 @@ def repository_params
       keys += [{:docker_tags_whitelist => []}, :docker_upstream_name] if params[:action] == 'create' || @repository&.docker?
       keys += [:ansible_collection_requirements, :ansible_collection_auth_url, :ansible_collection_auth_token] if params[:action] == 'create' || @repository&.ansible_collection?
       keys += [:label, :content_type] if params[:action] == "create"
+
+      if params[:action] == 'create' || params[:action] == 'update' || @repository&.generic?
```

Comment:
do we really need the  OR '@repository&.generic?'   

Should it be  "&& @repository&.generic?"

Reply:
with () around the params[:create]/params[:update]  part